File: Sway3i/src/main/java/com/sway3i/entities/TeacherDemand.java
package com.sway3i.entities;

import com.sway3i.entities.enums.DemandStatus;
import com.sway3i.entities.enums.EducationLevel;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Builder
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
public class TeacherDemand {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "created_by_id", nullable = false)
    private User createdBy;

    private String subject;

    @Enumerated(EnumType.STRING)
    private EducationLevel educationLevel;

    private String description;

    @Enumerated(EnumType.STRING)
    private DemandStatus status;
}

File: Sway3i-FrontEnd/src/app/modal/entities/teacher-demand.ts
import { DemandStatus } from '../enums/demand-status-enum';
import { EducationLevel } from '../enums/education-level-enum';
import { User } from './user';

export interface TeacherDemand {
  id: number;
  createdBy: User;
  subject: string;
  educationLevel: EducationLevel;
  description: string;
  status: DemandStatus;
}
